tests/e2e/test_pv_scale_ocs_create_delete_pvcs.py

```diff
@@ -0,0 +1,338 @@
+"""
+PV Create/Delete & Memory Leak Test: Test the PVC limit with 3 worker nodes
+with & without any IO create and delete the PVCs and check for memory leak
+TO DO: This Test needs to be executed in Scaled setup,
+Adding node scale is yet to be supported.
+"""
+import logging
+import pytest
+import time
+import random
+
+from concurrent.futures import ThreadPoolExecutor
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.exceptions import TimeoutExpiredError
+from tests import helpers
+from ocs_ci.framework.testlib import scale, E2ETest
+from tests.fixtures import (
+    create_ceph_block_pool, create_rbd_secret,
+    create_project, create_cephfs_secret
+)
+
+log = logging.getLogger(__name__)
+
+
+class BasePvcPodCreateDelete(E2ETest):
+    """
+    Base Class to create/delete PVC and POD
+    """
+    all_pvc_obj = []
+    all_pod_obj = []
+    delete_pod_count = 0
+    test_case = ""
+    timeout = 300
+
+    def pre_req(self):
+        """
+        pre_req def to calculate the median value for the memory leak
+        """
+        log.info(f"Pre-req to eveluate memory leak data "
+                 f"before start of test case")
+        self.median_dict = helpers.memory_leak_median()
+        log.info(f"Median dict values for memory leak {self.median_dict}")
+
+    def create_pvc_pod(self, rbd_sc_obj, cephfs_sc_obj, number_of_pvc, size):
+        """
+        def to create multiple PVC of different type and created pods on them.
+
+        Args:
+            rbd_sc_obj (obj_dict): rbd storageclass object
+            cephfs_sc_obj (obj_dict): cephfs storageclass object
+            number_of_pvc (int): pvc count to be created for each types
+            size (str): size of each pvc to be created eg: '10Gi'
+        """
+        access_modes = [constants.ACCESS_MODE_RWO, constants.ACCESS_MODE_RWX]
+        # Lists to evaluate the pod and pvc created during function call
+        temp_pod_objs, temp_pvc_objs = ([] for i in range(2))
+        future_cephfs_rwo_obj, future_cephfs_rwx_obj = ([] for i in range(2))
+        future_rbd_rwo_obj, future_rbd_rwx_obj = ([] for i in range(2))
+        future_pod_objs, future_temp_pod_objs = ([] for i in range(2))
+        # Flatten to remove the sublist created during future obj
+        flatten = lambda l: [item for sublist in l for item in sublist]
```

Comment:
Observing pep8 failure in flatten line,, any suggestions how to avoid this ?
In testcase using threads which returns future objects which will be list of lists, so to converge all the lists into one list using this flatten as in line 81, 83, etc.

Reply:
https://stackoverflow.com/questions/25010167/e731-do-not-assign-a-lambda-expression-use-a-def

Reply:
feel free to add
```
flatten = lambda l: [item for sublist in l for item in sublist]  # noqa: E731
```

Reply:
let me update this, Thanks Raz

Reply:
marking this as resolved